src/NuGet.Core/NuGet.Protocol/Utility/LocalFolderUtility.cs

```diff
@@ -987,18 +987,28 @@ private static string EnsurePackageExtension(string packagePath, bool isSnupkg)
         }
 
         /// <summary>
-        /// This method checks of a given list of package paths exist on disk.
+        /// If there isn't at least one Path specified, throw that no file paths were resolved for this Package.
         /// </summary>
-        /// <param name="packagePath">A package path to be used while creating logs.</param>
-        /// <param name="matchingPackagePaths">A list of matching package paths that need to be checked.</param>
+        /// <param name="packagePath">The package path the user originally provided for use while creating logs.</param>
+        /// <param name="matchingPackagePaths">A list of matching package paths that were previously resolved.</param>
         public static void EnsurePackageFileExists(string packagePath, IEnumerable<string> matchingPackagePaths)
         {
-            if (!matchingPackagePaths.Any())
+            if (!ResolvedAnyPackagePath(matchingPackagePaths))
             {
+                ErrorFileNotFound(packagePath);
+            }
+        }
+
+        public static void ErrorFileNotFound(string packagePath)
```

Comment:
```suggestion
        public static void ThrowUnableToFindFileArgumentException(string packagePath)
```

Reply:
@nkolev92 please comment whether adding "Throw" back into this method name is acceptable, since I just removed it at your request in the last commit.

Reply:
My suggestion is to not have this method at all. 
See https://github.com/NuGet/NuGet.Client/pull/3140#discussion_r355225439

Reply:
Ok, my takeaway here is to remove this method, and copy the throw logic to the 2-3 places it's needed.

Reply:
FWIW, I suggested using the word throw in the method name because it always throws, so anyone reading code that calls this method is more likely to have the expectation that the method will not return. The name "ErrorFileNotFound" sounds to me like it logs an error, so anything calling it still needs to return or throw, to avoid codepaths that expect the thing to have been found. And I very much objected to "FileNotFound" in the name, because it doesn't throw a `FileNotFoundException`.

But yes, I think "duplicating" the code to throw is better as it's more clear what the program flow is. I can see some value in a helper method because if `Strings.UnableToFindFile` ever changes the number of parameters, it's easier to catch all usage if it's in a single helper method, but I don't feel like `LocalFolderUtility` would be the best place for that. Maybe a `MessageUtility` class, but only for the string formatting part, the throw statement should be in the method that detected the error, not the utility method. I'd just go with duplicating it in the few places it's needed.